src/main/scala/io/iohk/ethereum/vm/Memory.scala

```diff
@@ -70,6 +70,21 @@ class Memory private(private val underlying: ByteString) {
       (newUnderlying.slice(start, end), new Memory(newUnderlying))
     }
 
+  /** This function will expand the Memory size as if storing data given the `offset` and `size`.
+    * If the memory is already initialised at that region it will not be modified, otherwise it will be filled with
+    * zeroes.
+    * This is required to satisfy memory expansion semantics for *CALL* opcodes.
+    */
+  def expand(offset: UInt256, size: UInt256): Memory = {
+    val totalSize = (offset + size).toInt
+    if (this.size >= totalSize || size.isZero)
```

Comment:
Yep, this is also a direct consequence of eq 216 (_M(s,f,l) = ..._). Good catch!